src/peft/peft_model.py

```diff
@@ -734,6 +734,10 @@ def forward(
     def generate(self, **kwargs):
         peft_config = self.active_peft_config
         self.base_model.prepare_inputs_for_generation = self.prepare_inputs_for_generation
+        if hasattr(self.base_model, "model"):
+            self.base_model.model.generation_config = self.generation_config
+        else:
+            self.base_model.generation_config = self.generation_config
```

Comment:
I think these changes are fine for now, I will investigate later why this change is needed